inst/rmarkdown/lua/codefolding.lua

```diff
@@ -0,0 +1,73 @@
+--[[
+codefolding - fold code blocks in HTML documents
+
+MIT License
+
+Copyright @ 2022-2022 Atsushi Yasumoto
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+--]]
+
+local DEFAULT_STATE
+
+function Meta(meta)
+  DEFAULT_STATE = meta["rmd_codefolding_lua"] == "show" and " open" or ""
+end
+
+function CodeBlock(elem)
+  local SUPPORTED_CLASSES = {
+    'r', 'python', 'bash', 'sql', 'cpp', 'stan', 'julia', 'foldable'
+  }
+
+  -- Do nothing if no supported classes are given
+  local folding = false
+  for _, v in pairs(SUPPORTED_CLASSES) do
+    if elem.classes:find(v) then
+      folding = true
+      break
+    end
+  end
+  if not folding then return end
+
+  -- Do nothin if has "fold-none" class
+  if elem.classes:find("fold-none") then return end
```

Comment:
You are right that `fold-none` is unsupported in `html_document`.
We can remove the feature from this PR or add the feature to `html_document`.
If we chose the latter, then I would work on it first.

Reply:
I just wanted to be sure to check I got it correctly, and so that we know we can improve later on `html_document()`. But we don't need to do it now. Really no problem to do it later, it can be hidden feature from some times.

Having separate PR would make more sense anyway. 

So if you think we can add to `html_document()` (which is a JS feature), then I suggest to do it later. Are you okay with this ?

Reply:
>I suggest to do it later

Alright, I will also consider adding some documents.

Reply:
I mark this comment resolved because I track this as an issue on #2346